cpp/tests/strings/integers_tests.cu

```diff
@@ -26,48 +26,188 @@
+                                      cudf::data_type{cudf::type_id::UINT32});
+  expected = cudf::test::fixed_width_column_wrapper<bool>({0, 0, 0, 1, 1, 1, 1, 1, 0});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected, true);
+
+  strings = cudf::test::strings_column_wrapper(
+    {"-9223372036854775809",    // std::numeric_limits<int64_t>::min() - 1
+     "-9223372036854775808",    // std::numeric_limits<int64_t>::min()
+     "-9223372036854775807",    // std::numeric_limits<int64_t>::min() + 1
+     "9223372036854775806",     // std::numeric_limits<int64_t>::max() - 1
+     "9223372036854775807",     // std::numeric_limits<int64_t>::max()
+     "9223372036854775808",     // std::numeric_limits<int64_t>::max() + 1
+     "18446744073709551614",    // std::numeric_limits<uint64_t>::max() - 1
+     "18446744073709551615",    // std::numeric_limits<uint64_t>::max()
+     "18446744073709551616"});  // std::numeric_limits<uint64_t>::max() + 1
+  results  = cudf::strings::is_integer(cudf::strings_column_view(strings),
+                                      cudf::data_type{cudf::type_id::INT64});
+  expected = cudf::test::fixed_width_column_wrapper<bool>({0, 1, 1, 1, 1, 0, 0, 0, 0});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected, true);
+
+  results  = cudf::strings::is_integer(cudf::strings_column_view(strings),
+                                      cudf::data_type{cudf::type_id::UINT64});
+  expected = cudf::test::fixed_width_column_wrapper<bool>({0, 0, 0, 1, 1, 1, 1, 1, 0});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected, true);
 }
 
 TEST_F(StringsConvertTest, ToInteger)
 {
-  std::vector<const char*> h_strings{
-    "eee", "1234", nullptr, "", "-9832", "93.24", "765é", "-1.78e+5", "2147483647", "-2147483648"};
+  std::vector<const char*> h_strings{"eee",
+                                     "1234",
+                                     nullptr,
+                                     "",
+                                     "-9832",
+                                     "93.24",
+                                     "765é",
+                                     nullptr,
+                                     "-1.78e+5",
+                                     "2147483647",
+                                     "-2147483648",
+                                     "2147483648"};
   cudf::test::strings_column_wrapper strings(
     h_strings.begin(),
     h_strings.end(),
     thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
-  std::vector<int32_t> h_expected{0, 1234, 0, 0, -9832, 93, 765, -1, 2147483647, -2147483648};
-
-  auto strings_view = cudf::strings_column_view(strings);
-  auto results = cudf::strings::to_integers(strings_view, cudf::data_type{cudf::type_id::INT32});
 
-  cudf::test::fixed_width_column_wrapper<int32_t> expected(
-    h_expected.begin(),
-    h_expected.end(),
-    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
```

Comment:
Can you change this back? I liked how the iterator used for nulls always matched based on the vector data.

Reply:
Done.